db.go

```diff
@@ -120,6 +120,13 @@ type DB struct {
 	// of truncate() and fsync() when growing the data file.
 	AllocSize int
 
+	TraceUpdateEnter func(*Tx, func(*Tx) error)
```

Comment:
I think bbolt should have some interceptor mechanism, but I think it should not be limited to tracing alone, 
but would allow to potentially execute additional logic, in some cases including additional modification to DB state. 

It would require detailed documentation of the semantic of the functions & tests. 

For consistency, please also see the discussion/naming aspects in: https://github.com/etcd-io/etcd/pull/12855